### PR TITLE
Update STANDARDS.md

### PR DESCRIPTION
**Pull request by amcisaac**
Updating the STANDARDS document to be in line with our current practices as well as outline things we'd like to ensure are included moving forward.

### CODE REVIEW

File: STANDARDS.md [hunk 1]
```diff
@@ -2,65 +2,69 @@
 This file outlines the standards and data requirements needed for submitting a dataset to QCArchive.
 This ensures that we have a consistent data model for downstream processes.
 
-STANDARDS version: 3 (adopted 2020.12.11)
+STANDARDS version: 4 (adopted 2024.11.XX)
```
Comment (amcisaac) [inline STANDARDS.md:5]: Question: do we want to update the version number? Nothing has ever conformed to version 3 as far as I can tell from the submissions directory, so we could continue using it, but I thought it might be better to bump it so there is no confusion.

File: STANDARDS.md [hunk 2]
```diff
@@ -2,65 +2,69 @@
...
+- A set of charges that the dataset contains
+- The min, mean and max molecular weight of molecules the dataset contains 
 
 ## Each revision must use the following procedure:
 
 A revision means creating a new dataset based on an existing one, with the intent of fixing/improving it.
 
-- The changelog must be copied from the current version, and a new entry added in the new version
-- The dataset version information is updated
+- The changelog must be copied from the current version (if it exists), and a new entry added for the new version. The changelog should describe any changes between the new version and previous version, including QCAIDs of any records that were removed or modified, with an explanation as to why they were removed.
```
Comment (amcisaac) [inline STANDARDS.md:61]: Are people happy with this description of a change log? The dictionary format sounded hard to parse and unnecessarily complicated. But I'm happy to put it back if people feel strongly.

File: STANDARDS.md [hunk 3]
```diff
@@ -2,65 +2,69 @@
...
+- The dataset version information is updated, and the description updated if necessary
 - The README is updated
-- A notebook or record of the lines of python used to manipulate the dataset responsible for the revision
-	- Each file of record should be named with the version that it first addresses, e.g. `submit-v3.0`
-		- If `submit-v3.0` also includes a `v3.1 and v3.2` update, then a new file for `v3.2.1` would be `submit-v3.2.1.py`
-	- Python notebooks should have version changes in order, and can be run incrementally.
+- A notebook or script used to manipulate the dataset and generate the revision
```
Comment (amcisaac) [inline STANDARDS.md:64]: Are revisions supposed to be submitted in their own directory, or in the directory of the original dataset? This makes it sound like the revision would be submitted in the directory of the original dataset. I don't think we're currently doing this, so we should be on the same page about it moving forward, and update the standards to indicate it explicitly.

File: STANDARDS.md [hunk 4]
```diff
@@ -70,7 +74,7 @@ A revision means creating a new dataset based on an existing one, with the inten
 
 ## Dataset naming and versioning
 
-Each dataset shall be versioned.
+Each dataset shall be versioned according to `vX.Y.Z`. X refers to the major version, Y to the minor version, and Z to a micro version.
 - The naming of a dataset should have the following structure:
 
     `"OpenFF <descriptive and uniquely-identifying name> v<version number>"`
```
Comment (amcisaac) [inline STANDARDS.md:80]: See question above about whether we want to update versioning to start at 4, or keep it at 3

File: STANDARDS.md [hunk 5]
```diff
@@ -79,17 +83,16 @@ Each dataset shall be versioned.
 
 * The major version shall indicate the STANDARDS that the dataset conforms to. Datasets which are not intended to conform to any STANDARDS should start with 0, e.g. `"v0.1"`. 
 
-* Datasets with versions starting with `"v1.x"` and `"v2.x"` do not follow any official STANDARDS, and thus should be considered `"v0.x"`.
+* Datasets with versions starting with `"v1.x"` and `"v2.x"` do not follow any official STANDARDS, and thus should be considered `"v0.x"`. 
 
 - A minor version change (e.g. `"v3.1"`) represents a minor addition and/or fixes problems:
-	- Adding molecules
+	- Adding molecules or removing/replacing records with an underlying problem
 	- Adding compute specifications
```
Comment (amcisaac) [inline STANDARDS.md:90]: Do we really want compute specifications to require a version bump? You already have to specifically request the compute spec you want, so I don't think there's any confusion. 

Additionally, our existing procedure for compute expansions doesn't allow modification of the dataset name, so we would have to modify or eliminate it in favor of submitting a completely new dataset for each compute spec update.

File: STANDARDS.md [hunk 6]
```diff
@@ -79,17 +83,16 @@ Each dataset shall be versioned.
...
 	- Errors/bugs in the molecule specification
 	- Changes necessary to adhere to the STANDARDS (i.e. changes necessary to placate the NONE compliance status)
 
 - A micro version change (e.g. `"v3.1.1"`) represents a cosmetic change, or a change that is based on dynamic information that does not change the underlying data:
```
Comment (amcisaac) [inline STANDARDS.md:94]: Given that we've eliminated the compute blacklist in favor of filtering out problems and bumping the version, do we want to use a micro version? I can't think of a cosmetic change that would require us to specify a version, for example updating the status or description wouldn't need a new version IMO

File: STANDARDS.md [hunk 7]
```diff
@@ -116,13 +119,13 @@ Pre-submission filtering:
 Post-submission filtering:
```
Comment (amcisaac) [inline STANDARDS.md:119]: Given the standards above for versions, doing any post-submission filtering would require a minor version bump. Is that how we intend this to be used? That would mean that any released dataset would be v3.1, and nothing would ever be v3.0 unless all the stereochemistry, topology, etc was perfect and didn't need to be filtered.

File: STANDARDS.md (hunk 1 above)
Comment (lilyminium) [reply STANDARDS.md:5]: SGTM!

File: STANDARDS.md (hunk 2 above)
Comment (lilyminium) [reply STANDARDS.md:61]: Based on its location in this doc, it sounded like the dict format for a CHANGELOG was intended to fit Pythonically in the metadata of the dataset somewhere. As a markdown file this makes sense to me.

Comment (lilyminium) [inline STANDARDS.md:61]:
```suggestion
- The changelog must be copied from the current version (if it exists), and a new entry added for the new version. The changelog should describe any changes between the new version and previous version, including QCArchive record ids of any records that were removed or modified, with an explanation as to why they were removed.
```

Just suggesting to explicitly refer to QCA record IDs instead of by shorthand :-)

**Review by lilyminium — APPROVED** [review]
LGTM!! I've left comments on the discussion points you raised.

**Comment by amcisaac** [comment]
OK great, thanks! I'll make these changes, and wait to merge until next week in case anyone else wants to comment

File: STANDARDS.md (hunk 7 above)
Comment (amcisaac) [reply STANDARDS.md:112]: Trevor mentioned that it's referring to the [`UnperceivableStereoFilter`](https://docs.openforcefield.org/projects/qcsubmit/en/stable/_modules/openff/qcsubmit/results/filters.html#UnperceivableStereoFilter), which writes the molecule to an SDF file and tries to read it back in. I think this has to be applied after an optimization, since the QM structure and the structure from the conformer generation may differ. But for a single point dataset, it could be done before.

I updated the text to mention that, and specify which other filters can be used.

File: STANDARDS.md [hunk 8]
```diff
@@ -115,14 +111,17 @@ Pre-submission filtering:
 
 Post-submission filtering:
 
-- OpenFF toolkit ingestion with strict stereo checking in RDKit
-- Hydrogen bonding
+- Stereochemistry is preserved (e.g. QCSubmit's `UnperceivableStereoFilter`)
+- CMILES (topology) change (e.g. QCSubmit's `ConnectivityFilter`)
+- Calculation ran successfully (e.g. QCSubmit's `RecordStatusFilter`)
+- Hydrogen bonding (for torsion drives only) (e.g. QCSubmit's `HydrogenBondFilter`)
```
Comment (ntBre) [inline STANDARDS.md:117]: It might be nice to link to the code or, likely preferably, the docs for these filters.

File: STANDARDS.md [hunk 9]
```diff
@@ -70,30 +72,24 @@ A revision means creating a new dataset based on an existing one, with the inten
...
 
-- The first submission of a dataset will have a version `"v3.0"`
+  Where the words in the dataset should be separated by spaces.
+
+- The first submission of a dataset will have a version `"v4.0"`
 
 * The major version shall indicate the STANDARDS that the dataset conforms to. Datasets which are not intended to conform to any STANDARDS should start with 0, e.g. `"v0.1"`. 
 
-* Datasets with versions starting with `"v1.x"` and `"v2.x"` do not follow any official STANDARDS, and thus should be considered `"v0.x"`.
+* Datasets with versions starting with `"v1.x"` and `"v2.x"` do not follow any official STANDARDS, and thus should be considered `"v0.x"`.
```
Comment (ntBre) [inline STANDARDS.md:86]: This is the tiniest complaint imaginable, but there's a trailing space here.

**Review by ntBre — APPROVED** [review]
Looks great to me, thanks for your work on this! I made two very small comments about the text, but these are obviously non-blocking and don't affect the actual standards, which look great.

File: STANDARDS.md (hunk 7 above)
Comment (lilyminium) [reply STANDARDS.md:112]: Thanks @amcisaac!
